server/auth_server.go

```diff
@@ -658,3 +657,61 @@ func IsSamlJWTValid(value string) (bool, map[string][]string) {
 	}
 	return false, samlData
 }
+
+func GetProvider(testAuthConfig model.TestAuthConfig) error {
+	authConfig := testAuthConfig.AuthConfig
+	newProvider, err := initProviderWithConfig(&authConfig)
+	if err != nil {
+		log.Errorf("GetProvider: Error initializing the provider %v", err)
+		return err
+	}
+	//store the config to db
+	log.Infof("newProvider %v", newProvider.GetName())
+	err = newProvider.TestLogin(&testAuthConfig)
+	if err != nil {
+		log.Errorf("GetProvider: Error in login %v", err)
+		return err
+	}
+
```

Comment:
I guess the above newProvider.TestLogin(&testAuthConfig) will check the auth call. What is the rest of the code for?